aten/src/ATen/native/cpu/ReduceAllOpsKernel.cpp

```diff
@@ -99,7 +99,7 @@ static void max_all_kernel_impl(Tensor& result, const Tensor& input) {
     reduce_all_impl<int64_t>(result, input, lower_bound<int64_t>(),
       [=](int64_t a, int64_t b) -> int64_t { return max_impl(a, b); });
   } else {
-    AT_DISPATCH_ALL_TYPES_AND_COMPLEX(input.scalar_type(), "max_all", [&] {
+    AT_DISPATCH_ALL_TYPES(input.scalar_type(), "max_all", [&] {
```

Comment:
nit: a test would be nice, especially to check that (1) we are raising a sane error message in this case and (2) the behavior is consistent between CPU and CUDA

Reply:
Thanks for your comment! I had later realized that I had forgotten to remove dtype checks & was going to submit another PR. 

`test_complex_unsupported()` exists to test condition 1. 

As for checking whether the behavior is consistent between CPU & CUDA, please advise if error messages of the exception for both should be compared. Thanks!

Reply:
> As for checking whether the behavior is consistent between CPU & CUDA, please advise if error messages of the exception for both should be compared. Thanks!

The error messages don't necessarily need to be the same but they should both be readable

Reply:
Thanks for your advice! [test_complex_unsupported ](https://github.com/pytorch/pytorch/blob/5546a12fe31eb3b2254eeb9837e4f1e6b000f256/test/test_torch.py#L5973) currently checks for RuntimeException. Could you please suggest how a test would check for readable error messages? For example, would you recommend checking if an error message has a length of, more than, say, 10? Or would you recommend checking if an error message contains a string such as _not implemented_ or _not support_ (_not support_ can also be present in _not supported_)? Thanks!

Reply:
Checking if the error message contains some substring seems like a good way to ensure the error message is understandable This can be done with `self.assertRaisesRegex`, e.g. https://github.com/pytorch/pytorch/blob/554a1a70c74aa1dfa89fbd469510092d86dd191b/test/test_torch.py#L115-L116

Reply:
> Checking if the error message contains some substring seems like a good way to ensure the error message is understandable This can be done with `self.assertRaisesRegex`, e.g.
> 
> https://github.com/pytorch/pytorch/blob/554a1a70c74aa1dfa89fbd469510092d86dd191b/test/test_torch.py#L115-L116

Thanks again for your help!
Yes, I was also going to use `self.assertRaisesRegex`, as I had earlier used it in `test_maximum_minimum_complex()` in [test_binary_ufuncs.py](https://github.com/imaginary-person/pytorch-1/blob/277d8e7ac1473dbe8be6c5fb4587205f4b9cfa4c/test/test_binary_ufuncs.py#L1068).